tests/vmi_multus_test.go

```diff
@@ -863,6 +865,92 @@ var _ = Describe("[Serial]SRIOV", func() {
 			Expect(libnet.PingFromVMConsole(vmi1, cidrToIP(cidrB))).To(Succeed())
 			Expect(libnet.PingFromVMConsole(vmi2, cidrToIP(cidrA))).To(Succeed())
 		})
+
+		Context("VLANed SR-IOV network with a vlaned VMI", func() {
+			var (
+				vlanedVMI1 *v1.VirtualMachineInstance
+				macVlaned1 string
+			)
+
+			const (
+				cidrVlaned1          = "192.168.0.1/24"
+				sriovVlanNetworkName = "sriov-vlan"
+			)
+
+			generateRandomMacAsString := func() string {
```

Comment:
I see no reason to use a lambda or even a helper function here.
You can place it in a `BeforeEach` directly and generate a pool of these (e.g. in a for-loop) so these can be used later on.

If you prefer a function, you can use a regular one.

Reply:
I mainly do not like the lambda here and the fact that is handles the error assertion.
The pool does not need to be managed more than it is now, e.g.:
```
var randomMac [3]string
---
for i := randomMac {
    randomMac[i], err := tests.GenerateRandomMac()
    Expect(err).ToNot(HaveOccurred())
}
```